src/main/java/pages/ListingListPage.java

```diff
@@ -0,0 +1,55 @@
+package pages;
+import features.Listing;
+import features.Search;
+import entities.DatabaseInteractor;
+
+import java.util.List;
+
+public class ListingListPage extends Page implements DatabaseInteractor{
+    private List<Listing> listings;
+
+    // super call from page, assume every page title will be listings?
+    public ListingListPage(List<Listing> listings) {
+        super("Listings");
+        this.listings = listings;
+    }
+    // getters and setters
+    public List<Listing> getListings() {
+        return listings;
+    }
+
+    public void setListings(List<Listing> listings) {
+        this.listings = listings;
+    }
+
+
+
+    // Implementing all methods from interface
+    // placeholder since I won't need this method
+    public void create() {
+        return;
+    }
+
+    public void update() {
+        return;
+    }
+    public void delete() {
+        return;
+    }
+
+    public T query() {
+        // TODO: Implement query which returns data, will have to access database which is not yet setup
+        return data;
+    }
```

Comment:
i know this part is still really unsure, but you will probs create a ListingListPageDatabaseInteractor which know how to manage database for this use case, i.g. the query method will return a list of listings matching whatever the search parameter is. you would implement these methods in that ListingListPageDatabaseInteractor, not in the page itself. atleast that's how i think it will work, but clare probs knows better lol

Reply:
so you can type the method like `public Listing[] query()` in the interactor you will write. i used T bc i used generics, bc the query method will return something different depending on the table it is working on. if it's getting Reviews, the return type might be Review[]. generics are just giving u the freedom to define your own return type for different use cases

Reply:
for now tho this is fine, i understand what you mean by this

Reply:
Yeah that makes sense. I'll start on the interactor between listing list and database soon